### DISCUSSION TITLE
Disable website icons setting was reset?

### DISCUSSION THREAD

**Discussion by sfkpmr** [dani-garcia/vaultwarden#2672] [Q&A] [answered]
### Subject of the issue
Back in May I [asked](https://github.com/dani-garcia/vaultwarden/discussions/2503) about how the website icons setting worked, and disabled it. Today I checked the logs and noticed that it had again downloaded icons, and in the UI I saw that all credentials had had icons downloaded.

### Deployment environment

* vaultwarden version: 1.25.0, now updated to 1.25.2

* Install method: Docker Compose

  vaultwarden:
    image: vaultwarden/server
    container_name: vaultwarden
    environment:
      - TZ=${TZ}
      - SIGNUPS_ALLOWED=false
    volumes:
      - ./vaultwarden:/data
    security_opt:
      - no-new-privileges:true
    restart: unless-stopped

* Clients used: Web, Android, Chromium add-on

* Reverse proxy and version: Caddy 2.5.1

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce
Disabled the website icons setting in May, now it's been reset.

### Expected behaviour
Settings don't change automatically.

### Actual behaviour
Setting changed on its own.

**Comment by BlackDex** [comment]
@sfkpmr may i ask where you changed these settings?
If this was in the web-vault, then those are stored in the local storage/cookies of the browser, and thus if you clear that, or change hosts, or use a different browser, then those settings will not be there. Same for the web-vault theme for example.

**Comment by sfkpmr** [comment]
@BlackDex Yes, I changed the settings at /#/settings/preferences in the web UI. 

So, is there no way of setting this permanently? 

I found [this](https://github.com/dani-garcia/vaultwarden/blob/main/.env.template), can I set DISABLE_ICON_DOWNLOAD=false as an environment variable? Or true in my case.

**Comment by BlackDex** [comment, marked as answer]
Yes, that will prevent at least trying to fetch those icons. It will not prevent the clients from requesting them to download, and thus those requests will still end up into the reverse proxy logs.

But they will not be cached or stored locally.

**Reply by sfkpmr** [reply]
Thanks, giving that a try now! 

I guess the icons themselves are stored in the client as well, not just the setting?

I made a new vault account with some entries to verify over the week.

**Reply by BlackDex** [reply]
Icons are cached by the clients as far as i know. But it may depend upon the client on how long unless they adhere to the HTTP headers which report a specific cache lifetime.

**Reply by sfkpmr** [reply]
Thanks, hopefully no more issues now! :)
